pandas/core/aggregation.py

```diff
@@ -384,3 +385,102 @@ def validate_func_kwargs(
     if not columns:
         raise TypeError(no_arg_message)
     return columns, func
+
+
+def transform(
+    obj: FrameOrSeries,
+    func: Union[str, List, Dict, Callable],
+    axis: int,
+    *args,
+    **kwargs,
+) -> FrameOrSeries:
+    """
+    Transform a DataFrame or Series
+
+    Parameters
+    ----------
+    obj : DataFrame or Series
+        Object to compute the transform on.
+    func : string, function, list, or dictionary
+        Function(s) to compute the transform with.
+    axis : {0 or 'index', 1 or 'columns'}
+        Axis along which the function is applied:
+
+        * 0 or 'index': apply function to each column.
+        * 1 or 'columns': apply function to each row.
+
+    Returns
+    -------
+    DataFrame or Series
+        Result of applying ``func`` along the given axis of the
+        Series or DataFrame.
+
+    Raises
+    ------
+    ValueError
+        If the transform function fails or does not transform.
+    """
+    is_series = obj.ndim == 1
+
+    if obj._get_axis_number(axis) == 1:
+        assert not is_series
```

Comment:
this is not user visible right?

Reply:
Correct - when obj is a Series and axis is 1 or 'columns', `obj._get_axis_number(axis)` will raise with the ValueError "No axis named columns for object type Series" (or 1).